app/src/main/java/org/greenstand/android/TreeTracker/activities/MainActivity.kt

```diff
@@ -232,4 +251,15 @@ class MainActivity : AppCompatActivity(), ActivityCompat.OnRequestPermissionsRes
         locationUpdateManager.startLocationUpdates()
         locationDataCapturer.start()
     }
+
+    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {
+        // Ignore
+    }
+
+    override fun onSensorChanged(event: SensorEvent?) {
```

Comment:
The [docs](https://developer.android.com/guide/topics/sensors/sensors_motion) suggest using the job scheduler to query the current step count.

Reply:
I did see that. But I saw this example in the documentation for `SensorManager` in android developer guide. I wish there was an example for using JobScheduler.

Reply:
I think it would be worth it to find out away to do this with the JobScheduler since it mentions it can drain the battery otherwise.

Reply:
I thought there would be an API like you are suggesting in the beginning. But there isn't one.  Actually, the only way to get a step count value is to listening to SensorEvent. At least to my knowledge. 

Thus, our use case needs this value at regular intervals. Of the three cadences available, `SENSOR_DELAY_NORMAL` seems more friendly to battery usage while maintaining a level of accuracy. The other choices are `SENSOR_DELAY_FASTEST`, `SENSOR_DELAY_GAME`, `SENSOR_DELAY_UI`. 

 I believe the current implementation won't be any more draining than a JobScheduler implementation since we are going to get frequent intervals of reading the value from the change event anyways.

Reply:
What I mean by the API suggestion is to make one like that ourselves to wrap what android supplies. We don't need constant updates of the step count, we only need the step count at the time that we plant the tree, correct?

So instead of getting a continuous stream of step updates, we can listen to step events until we get one step update, then we unsubscribe and use that value for the calculation.

The current implementation will get step updates even if the user is just looking at the map or syncing data.

Reply:
I made a change now that uses API to enable the step count listener and collect the values at the time of interest and disable the listener. In addition created an observer on destroy lifecyle event to unregister the listener when the app is exited without completing the cycle enabling/disabling the listener.